app/services/round_timer.rb

```diff
@@ -0,0 +1,88 @@
+class RoundTimer
+
+  def initialize(round)
+    @round = round
+  end
+
+  def start!
+    round_timer_activations.create! start_time: Time.current
+  end
+
+  def stop!
+    round_timer_activations.last&.update(stop_time: Time.current)
+  end
+
+  def reset!
+    round_timer_activations.destroy_all
+  end
+
+  def show?
+    round_timer_activations.count > 0 && !completed?
+  end
+
+  def paused?
```

Comment:
Can the States you've defined below be used here instead of the poking around on last?

Reply:
I'm not sure if there's a good way to do it. The state function needs to check if it's paused first before it knows what else needs to be calculated. In order to know if it's paused it needs to add up all the previous activations. It's not a lot of computation since there won't be many activations, but it would be nice to feel like it ties together a bit better.